TenantAPI/src/main/java/com/qa/tenantApi/main/service/TenantService.java

```diff
@@ -0,0 +1,73 @@
+package com.qa.tenantApi.main.service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+
+import com.qa.tenantApi.main.Constants;
+import com.qa.tenantApi.main.entities.Tenant;
+import com.qa.tenantApi.main.repository.TenantRepo;
+
+@Service
+public class TenantService {
+
+	private TenantRepo tenantRepo;
+
+	public TenantService(TenantRepo tenantRepo) {
+		this.tenantRepo = tenantRepo; 
+	}
+
+	public List<Tenant> getAllTenants() {
+		return this.tenantRepo.findAll();
+	}
+	
+	public List<Tenant> tenantSearch(Tenant tenant){
+		return this.getAllTenants().stream().filter(x->x.matches(tenant)).collect(Collectors.toList());
+	}
+	
+	public String createTenant(Tenant tenant) {
+		this.tenantRepo.save(tenant);
+		return Constants.getCreationMessage();
+	}
+	
+	public String deleteTenant(Tenant tenant) {
+		this.tenantRepo.delete(tenant);
+		return Constants.getDeletionMessage();
+	}
+	
+	public String deleteTenantGroup(List<Tenant> tenants) {
+		this.tenantRepo.deleteAll(tenants);
+		return Constants.getGroupDeletionMessage();
+	}
+	
+	public String deleteAllTenants() {
+		this.tenantRepo.deleteAll();
+		return Constants.getAllDeletionMessage();
+	}
+	
+	public String updateTenant(Long id, Tenant updateTenant) {
+		Tenant tenantToUpdate = this.tenantRepo.findById(id).orElse(new Tenant());
+		tenantToUpdate.setFirstName(Optional.ofNullable(updateTenant.getFirstName()).orElse(Optional.ofNullable(tenantToUpdate.getFirstName()).orElse(Constants.getNaString())));
+		tenantToUpdate.setLastName(Optional.ofNullable(updateTenant.getLastName()).orElse(Optional.ofNullable(tenantToUpdate.getLastName()).orElse(Constants.getNaString())));

```

Comment:
This functionality should be extracted to its own method.

Reply:
Done